src/main/java/slimeknights/tconstruct/plugin/crt/managers/TinkerStationManager.java

```diff
@@ -0,0 +1,99 @@
+package slimeknights.tconstruct.plugin.crt.managers;
+
+import com.blamejared.crafttweaker.api.CraftTweakerAPI;
+import com.blamejared.crafttweaker.api.annotations.ZenRegister;
+import com.blamejared.crafttweaker.api.data.IData;
+import com.blamejared.crafttweaker.api.item.IIngredient;
+import com.blamejared.crafttweaker.api.item.IIngredientWithAmount;
+import com.blamejared.crafttweaker.api.item.IItemStack;
+import com.blamejared.crafttweaker.api.managers.IRecipeManager;
+import com.blamejared.crafttweaker.impl.actions.recipes.ActionAddRecipe;
+import net.minecraft.item.crafting.IRecipeType;
+import net.minecraft.util.ResourceLocation;
+import org.openzen.zencode.java.ZenCodeType;
+import slimeknights.mantle.recipe.SizedIngredient;
+import slimeknights.tconstruct.library.modifiers.Modifier;
+import slimeknights.tconstruct.library.modifiers.ModifierEntry;
+import slimeknights.tconstruct.library.recipe.RecipeTypes;
+import slimeknights.tconstruct.library.recipe.tinkerstation.ITinkerStationRecipe;
+import slimeknights.tconstruct.library.recipe.tinkerstation.modifier.IncrementalModifierRecipe;
+import slimeknights.tconstruct.library.recipe.tinkerstation.modifier.ModifierMatch;
+import slimeknights.tconstruct.library.recipe.tinkerstation.modifier.ModifierRecipe;
+import slimeknights.tconstruct.library.recipe.tinkerstation.modifier.OverslimeModifierRecipe;
+import slimeknights.tconstruct.plugin.crt.CRTHelper;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Collectors;
+
+@ZenRegister
+@ZenCodeType.Name("mods.tconstruct.TinkerStation")
+public class TinkerStationManager implements IRecipeManager {
+  
+  @ZenCodeType.Method
+  public void addOverslimeModifierRecipe(String name, IIngredient ingredient, int restoreAmount) {
+    name = fixRecipeName(name);
+    ResourceLocation id = new ResourceLocation("crafttweaker", name);
+    OverslimeModifierRecipe recipe = new OverslimeModifierRecipe(id, ingredient.asVanillaIngredient(), restoreAmount);
+    CraftTweakerAPI.apply(new ActionAddRecipe(this, recipe));
+  }
+  
+  @ZenCodeType.Method
+  public void addModifierRecipe(String name, IIngredientWithAmount[] inputs, IIngredient toolRequired, String modifierResult, int modifierResultLevel, int maxLevel, int upgradeSlots, int abilitySlots) {
```

Comment:
`upgradeSlots` and `abilitySlots` are mutually exclusive, setting both would cause problems with JEI and a few future features.

I would suggest possibly making 2 methods: one for upgrades and one for abilities. I suppose a third for slotless modifiers would make sense, but you can just call either of the other 2 with 0 as the parameter.